addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,279 @@
+
+    def _check_filters_results(self, post, survey_id, lines_domain):
+
+        with self.assertQueryCount(1):
+            base_domain = Survey._get_results_page_domain(survey_id, **post)
+            expected_user_input_line_ids = Survey._get_matching_user_input_lines(self, lines_domain, base_domain)
+
+        with self.assertQueryCount(1):
+            found_user_input_line_ids, _ = Survey._extract_filters_data(self, survey_id, post)
+
+        self.assertEqual(expected_user_input_line_ids, found_user_input_line_ids)
+
+    def test_get_matching_user_input_lines(self):
+        """
+            Check that we retrieve the accurate user input lines depending on the domain
+            Filters:
+                A,68,70 (matrix) : How often should you water those plants | Cactus : Once a month
+                A,0,57  (simple_choice) : Which of the following would you use to pollinate : Bees
+                L,0,107 (char_box) : What is your email ? : mark.brown23@example.com
+        """
+        user_input_domain = expression.AND([[('survey_id', '=', self.quiz_survey_id)], [('test_entry', '=', False)]])
+        filters_domain_items = [
+            expression.AND([[('matrix_row_id', '=', 68)], [('suggested_answer_id', '=', 70)], [('question_id', '=', 41)]]),
+            expression.AND([[('suggested_answer_id', '=', 57)], [('question_id', '=', 38)]]),
+            expression.AND([[('value_char_box', '=', 'mark.brown23@example.com')], [('question_id', '=', 28)]])
+        ]
+        found_user_input_lines = Survey._get_matching_user_input_lines(self, filters_domain_items, user_input_domain)
+        expected_user_input_lines = self.env['survey.user_input.line'].search([('user_input_id', '=', 9)])
+
+        self.assertEqual(found_user_input_lines, expected_user_input_lines)
+
+    def test_get_filters_from_post(self):
+        """
+            Check that we correctly retrieve the filters from the post
+        """
+        post = {'filters': 'A,14,101|A,0,58|L,0,2'}
+        answer_cols_to_rows_mapping, answer_user_input_line_ids = Survey._get_filters_from_post(post)
+        self.assertEqual(answer_cols_to_rows_mapping, {101: [14], 58: [0]})
+        self.assertEqual(answer_user_input_line_ids, [2])
+
+        post = {'filters': 'L,4,2'}
+        # There cannot be any row_id for a 'survey.user_input.line' model filter, the filter should be ignored
+        answer_cols_to_rows_mapping, answer_user_input_line_ids = Survey._get_filters_from_post(post)
+        self.assertFalse(answer_cols_to_rows_mapping)
+        self.assertFalse(answer_user_input_line_ids)
+
+        post = {'filters': 'A,0,9|J,40,3'}
+        # No model associated with the J letter, the second filter should be ignored
+        answer_cols_to_rows_mapping, answer_user_input_line_ids = Survey._get_filters_from_post(post)
+        self.assertEqual(answer_cols_to_rows_mapping, {9: [0]})
+        self.assertFalse(answer_user_input_line_ids)
+
```

Comment:
I'm not sure it is worth testing the ignoring fallback behavior. I think it would make more sense if we raised an error of some kind and wanted to make sure the user gets it, but as we don't think (so far) that it is useful either, I would drop this part.